code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -1520,143 +1520,101 @@
-			hive = hive_datum[X.hivenumber]
-		else
-			X.hivenumber = XENO_HIVE_NORMAL
-			hive = hive_datum[X.hivenumber]
+	var/xenoinfo = ""
+	var/can_overwatch = FALSE
 
-		var/leader = ""
+	var/tier3counts = ""
+	var/tier2counts = ""
+	var/tier1counts = ""
 
-		if(X in hive.xeno_leader_list)
-			leader = "<b>(-L-)</b>"
+	if(isxenoqueen(user))
+		var/mob/living/carbon/Xenomorph/Queen/Q = user
+		if(Q.ovipositor)
+			can_overwatch = TRUE
+
+	for(var/i in hive.xenos_by_typepath[/mob/living/carbon/Xenomorph/Queen])
+		var/mob/living/carbon/Xenomorph/X = i
+		xenoinfo += "<tr><td>[X.name] "
+		if(!X.client)
+			xenoinfo += " <i>(SSD)</i>"
+		else if(X.client.prefs.xeno_name && X.client.prefs.xeno_name != "Undefined")
+			xenoinfo += "- [X.client.prefs.xeno_name]"
+		var/area/A = get_area(X)
+		xenoinfo += " <b><font color=green>([A ? A.name : null])</b></td></tr>"
 
-		var/xenoinfo
-		if(user && anchored && X != user)
-			xenoinfo = "<tr><td>[leader]<a href=?src=\ref[user];watch_xeno_number=[X.nicknumber]>[X.name]</a> "
+	for(var/i in hive.xeno_leader_list)
+		var/mob/living/carbon/Xenomorph/X = i
+		if(can_overwatch)
+			xenoinfo += "<tr><td><b>(-L-)</b><a href=?src=\ref[user];watch_xeno_number=[X.nicknumber]>[X.name]</a> "
 		else
-			xenoinfo = "<tr><td>[leader][X.name] "
+			xenoinfo += "<tr><td><b>(-L-)</b>[X.name] "
 		if(!X.client)
 			xenoinfo += " <i>(SSD)</i>"
 		else if(X.client.prefs.xeno_name && X.client.prefs.xeno_name != "Undefined")
 			xenoinfo += "- [X.client.prefs.xeno_name]"
 
-		count++ //Dead players shouldn't be on this list
+		var/area/A = get_area(X)
+		xenoinfo += " <b><font color=green>([A ? A.name : null])</b></td></tr>"
+
+	for(var/typepath in hive.xenos_by_typepath)
+		var/mob/living/carbon/Xenomorph/T = typepath
+		if(initial(T.tier) == XENO_TIER_ZERO)
+			continue
+		if(typepath == /mob/living/carbon/Xenomorph/Queen)
+			continue
+		
+		switch(initial(T.tier))
+			if(XENO_TIER_THREE)
+				tier3counts += " | [initial(T.name)]s: [length(hive.xenos_by_typepath[typepath])]"
+			if(XENO_TIER_TWO)
+				tier2counts += " | [initial(T.name)]s: [length(hive.xenos_by_typepath[typepath])]"
+			if(XENO_TIER_ONE)
+				tier1counts += " | [initial(T.name)]s: [length(hive.xenos_by_typepath[typepath])]"
+
+		for(var/i in hive.xenos_by_typepath[typepath])
+			var/mob/living/carbon/Xenomorph/X = i
+			if(X.queen_chosen_lead)
+				continue
+			if(can_overwatch)
+				xenoinfo += "<tr><td><a href=?src=\ref[user];watch_xeno_number=[X.nicknumber]>[X.name]</a> "
+			else
+				xenoinfo += "<tr><td>[X.name] "
+			if(!X.client)
+				xenoinfo += " <i>(SSD)</i>"
+			else if(X.client.prefs.xeno_name && X.client.prefs.xeno_name != "Undefined")
+				xenoinfo += "- [X.client.prefs.xeno_name]"
+			var/area/A = get_area(X)
+			xenoinfo += " <b><font color=green>([A ? A.name : null])</b></td></tr>"
+
+	for(var/i in hive.xenos_by_typepath[/mob/living/carbon/Xenomorph/Larva])
+		var/mob/living/carbon/Xenomorph/X = i
```

Comment:
This span of code is done 4 times if i'm not mistaken, making it a `#define` or a proc would reduce code duplication.

Reply:
good idea.